jobs: add stub implementation of consistency check job

spilchen opened the pull request:
This adds infrastructure for a new job that will handle data consistency checking. It will be initially invoked via the EXPERIMENTAL SCRUB command and a new session variable (enable_scrub_job). It will eventually be invoked under its own SQL, but that will be handled separately.

I split this into 4 commits to make reviewing easier. I will squash them prior to merging to master.

## jobs: add stub implementation of consistency check job

This commit introduces a stub for the new CONSISTENCY_CHECK job type. The Resumer doesn't do anything useful yet. It simply returns as a no-op.

Informs: #148289
Epic: CRDB-30356
Release note: None

## sql: support starting consistency check job from SCRUB

This change adds support for initiating a CONSISTENCY_CHECK job when running SCRUB TABLE, controlled by the session variable enable_scrub_job.

When enable_scrub_job is true, SCRUB TABLE will schedule or start a consistency check job instead of running inline checks.

Informs: #148289
Epic: CRDB-30356
Release note: None

##  sql: add logic tests for consistency check job via SCRUB

This change introduces logic tests for the new consistency check job path when enable_scrub_job is enabled.

Informs: #148289
Epic: CRDB-30356
Release note: None

## sql: add unit test for SCRUB job execution semantics

This change adds a unit test (TestCheckJobImplicitTxnSemantics) to verify the behavior of consistency check jobs started via EXPERIMENTAL SCRUB TABLE in implicit transactions.

Includes test hooks via ConsistencyCheckTestingKnobs to simulate and control job execution behavior.

Informs: #148289
Epic: CRDB-30356
Release note: None

spilchen committed:
jobs: add stub implementation of consistency check job

This commit introduces a stub for the new CONSISTENCY_CHECK job type.
The Resumer doesn't do anything useful yet. It simply returns as a
no-op.

Informs: #148289
Epic: CRDB-30356
Release note: None

spilchen committed:
sql: support starting consistency check job from SCRUB

This change adds support for initiating a CONSISTENCY_CHECK job when
running SCRUB TABLE, controlled by the session variable
enable_scrub_job.

When enable_scrub_job is true, SCRUB TABLE will schedule or start a
consistency check job instead of running inline checks.

Informs: #148289
Epic: CRDB-30356
Release note: None

spilchen committed:
sql: add logic tests for consistency check job via SCRUB

This change introduces logic tests for the new consistency check job
path when enable_scrub_job is enabled.

Informs: #148289
Epic: CRDB-30356
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/148614)
<!-- Reviewable:end -->

spilchen committed:
sql: add unit test for SCRUB job execution semantics

This change adds a unit test (TestCheckJobImplicitTxnSemantics) to
verify the behavior of consistency check jobs started via EXPERIMENTAL
SCRUB TABLE in implicit transactions.

Includes test hooks via ConsistencyCheckTestingKnobs to simulate and
control job execution behavior.

Informs: #148289
Epic: CRDB-30356
Release note: None
